### PR TITLE
Add reduced prop to CardSearchResult to conditionally render search results compacted

### PR DESCRIPTION
**Pull request by anthonyshibitov**
### Contributor checklist

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Add reduced prop to CardSearchResult to conditionally render search results compacted

Notee: there's seem the be a bug with the date props. Even though they are marked as optional in the TS definition files, if they are not defined the component will break and not render.

### Related issue

- #601

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks, @anthonyshibitov! 🙏 The frontend formatting seems to be [failing on this](https://github.com/activist-org/activist/actions/runs/7215943446/job/19661175288?pr=607). Do you want to send along changes for this, or should I go ahead and do the full review? :)

**Comment by anthonyshibitov** [comment]
> Thanks, @anthonyshibitov! 🙏 The frontend formatting seems to be [failing on this](https://github.com/activist-org/activist/actions/runs/7215943446/job/19661175288?pr=607). Do you want to send along changes for this, or should I go ahead and do the full review? :)

all set for review!

**Comment by andrewtavis** [comment]
Thanks @anthonyshibitov! Will check it tomorrow :)

File: frontend/components/card/search-result/CardSearchResultOrganization.vue [hunk 1]
```diff
@@ -2,11 +2,11 @@
   <div class="flex flex-col w-full md:flex-row">
     <div class="flex justify-center w-full md:w-fit">
       <div
-        class="border rounded-lg w-fit border-light-section-div dark:border-dark-section-div bg-light-content"
+        class="border rounded-lg w-fit border-light-section-div dark:border-dark-section-div bg-light-content dark:bg-dark-content"
```
Comment (andrewtavis) [inline frontend/components/card/search-result/CardSearchResultOrganization.vue:5]: This isn't needed as we want it to be white for now so the back tech from below logo shows up, but also no stress 😊

File: frontend/components/card/search-result/CardSearchResultUser.vue [hunk 2]
```diff
@@ -54,6 +54,7 @@ import type { User } from "~/types/user";
 
 defineProps<{
   isPrivate?: boolean;
+  reduced?: boolean;
```
Comment (andrewtavis) [inline frontend/components/card/search-result/CardSearchResultUser.vue:57]: Looks like the changes weren't applied to user search results? I can get these :)

**Review by andrewtavis — APPROVED** [review]
All looking good, @anthonyshibitov! I'll bring it in and do a quick commit to get the user reactivity in there 😊

**Comment by andrewtavis** [comment]
And thanks for the formatting fixes! Hope that the pre-commit process was ok 😊

**Pull request merged**